src/tests/ftest/util/nvme_utils.py

```diff
@@ -208,7 +208,8 @@ def set_device_faulty(self, server, disk_id):
         self.dmg.storage_set_faulty(disk_id)
         result = self.dmg.storage_query_device_health(disk_id)
         # Check if device state changed to EVICTED.
-        if 'State:EVICTED' not in result.stdout_text:
+        if result['response']['host_storage_map']['storage']['smd_info']['devices']['dev_state'] \
+                != 'EVICTED':
```

Comment:
I think it's recommended to do something like this:
https://github.com/daos-stack/daos/blob/dc0beb982a9be6f190f0819c25c5655a044cf9ff/src/tests/ftest/nvme/health.py#L115

@phender Would know better